addons/sourcemod/scripting/freak_fortress_2.sp

```diff
@@ -4405,9 +4406,41 @@ public OnClientPutInServer(client)
 
 public OnClientDisconnect(client)
 {
-	if(Enabled && IsClientInGame(client) && IsPlayerAlive(client) && CheckRoundState()==1)
+	if(Enabled)
 	{
-		CreateTimer(0.1, CheckAlivePlayers);
+		if(IsBoss(client) && cvarPreroundBossDisconnect)
+		{
+			if (CheckRoundState()==1)
+			{
+				ForceTeamWin(OtherTeam);
+			}
+			
+			if (CheckRoundState()<=0)
+			{
+				new bool:omit[MaxClients+1];
+				Boss[0]=GetClientWithMostQueuePoints(omit);
```

Comment:
Also, the person who's disconnecting might not be `Boss[0]`-for example, if they're a companion.
You might want to do `Boss[GetBossIndex(client)]` here instead.